nodejs/restify/db.js

```diff
@@ -0,0 +1,40 @@
+const bcrypt     = require('bcrypt');
+const odbc       = require('odbc');
+const saltRounds = 12;
+
+// This example runs on IBM i using *LOCAL credentials. This will use the 
+// user profile of whoever runs the server, including their authorization lists.
+// If you want to run this example on another system and connect to IBM i, need
+// to change the connection string to connect remotely!
+let pool = odbc.pool('DSN=*LOCAL');
+
+setupDatabase();
+
+async function setupDatabase() {
+
+  // for this example to work in a contained manner, need to set up our Db2 for
+  // i schema by creating a table to hold our Books and a table to hold User
+  // information (username and password hash).
+  try {
+    let localPool = await pool;
+
+    await localPool.query("CREATE OR REPLACE TABLE BOOKS(ID INTEGER NOT NULL GENERATED ALWAYS AS IDENTITY (START WITH 1 INCREMENT BY 1), TITLE VARCHAR(32) NOT NULL, AUTHOR VARCHAR(32), PRIMARY KEY (ID))");
+
+    await localPool.query("CREATE OR REPLACE TABLE USERS(USERNAME VARCHAR(32) NOT NULL, PASSWORD_HASH VARCHAR(128) NOT NULL)");
```

Comment:
According to the docs the hash only requires a 60 characters long field https://www.npmjs.com/package/bcrypt#hash-info.

Reply:
Ah, I gave a really cursory glance but couldn't find it. Thanks!